daemon/container_unix.go

```diff
@@ -1054,29 +1140,76 @@ func (container *Container) releaseNetwork() {
 		return
 	}
 
-	n, err := container.daemon.netController.NetworkByID(nid)
-	if err != nil {
-		logrus.Errorf("error locating network id %s: %v", nid, err)
-		return
-	}
+	controller := container.daemon.netController
 
-	ep, err := n.EndpointByID(eid)
-	if err != nil {
-		logrus.Errorf("error locating endpoint id %s: %v", eid, err)
-		return
+	for _, ns := range networks {
+		n, err := findNetwork(controller, ns)
+		if err != nil {
+			continue
+		}
+		container.disconnectFromNetwork(n, false)
```

Comment:
I don't understand what's wrong with having `disconnectFromNetwork` do `updateSettings` in all cases :-/ What makes this code path special?